### PR TITLE
Homework 5(oop)

### PR DESCRIPTION
**Pull request by HunkevychPhilip**

### CODE REVIEW

File: src/oop/classes/Cookie.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public $placeholder; //cookie array
+
+    public function __construct($cookie)
+    {   
+        $this->placeholder = $cookie;
+    }
+
+    public function get($key, $default = null)
+    {
+        if(!empty($this->placeholder[$key])){
```
Comment (yderevianko-cartera) [inline src/oop/classes/Cookie.php]: should be spaces after and before if

File: src/oop/classes/Cookie.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        if(!empty($only)){
+            foreach ($only as $filteringKey) {
+                    if(isExists($filteringKey, $this->placeholder)){
+                        array_push($result, $this->placeholder[$filteringKey]);
+                    }
+                }
+            }
+
+        if(!empty($result)){
```
Comment (yderevianko-cartera) [inline src/oop/classes/Cookie.php]: spaces

File: src/oop/classes/Session.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    public function __construct($session)
+    {
+        $this->placeholder = $session;
+    }
+
+    public function set($key, $value = 'default')
+    {
+        $this->placeholder = array_merge($this->placeholder, [$key => $value]);
+        return "Data with the $value value was created under the $key key.";
```
Comment (yderevianko-cartera) [inline src/oop/classes/Session.php]: it is better to trow exception

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]
Please write unit tests

File: src/oop/classes/Session.php (hunk 3 above)
Comment (HunkevychPhilip) [reply src/oop/classes/Session.php]: I made this string-return only for the method's working visualization on the web page.
Should I still replace it by throwing an exception or just delete the return statement at all?

Comment (yderevianko-cartera) [reply src/oop/classes/Session.php]: You should always separate logic and representation and one class should not have both. That is why you should not use strings as return

File: src/oop/classes/Request.php [hunk 4]
```diff
@@ -5,9 +5,17 @@ class Request
     public $query;
     public $request;
     public $server;
+    public $session;
+    public $cookie;
 
     public function __construct()
     {
+        include_once 'Cookie.php';
+        include_once 'Session.php';
```
Comment (yderevianko-cartera) [inline src/oop/classes/Request.php]: please never include files inside constructor

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: tests/OOP-homework/SessionTest.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        $this->expectException(TypeError::class);
+        $this->fixture->session->all(23.4);
+    }
+
+    public function testAllDefault()
+    {
+        $this->assertEquals(['test1' => 'session_value', 'test2' => 'another_session_value'],
+            $this->fixture->session->all());
+    }
+    // <<<<< 'all' METHOD's TESTS END
```
Comment (yderevianko-cartera) [inline tests/OOP-homework/SessionTest.php]: please do not add comments like START and END of method

File: src/oop/index.php [hunk 6]
```diff
@@ -1,23 +1,20 @@
 <?php
 
-require_once "classes/Request.php";
-require_once "classes/Cookie.php";
-require_once "classes/Session.php";
-require_once "functions.php";
+namespace src\classes;
```
Comment (yderevianko-cartera) [inline src/oop/index.php]: namespace should be added only to classes/interface and not to templates.

File: src/oop/pages/cookie.php [hunk 7]
```diff
@@ -1,10 +1,12 @@
 <?php
 
-require_once '../functions.php';
-require_once '../classes/Request.php';
+namespace src\classes;
```
Comment (yderevianko-cartera) [inline src/oop/pages/cookie.php]: namespace should be added only to classes/interface and not to templates.

File: src/oop/pages/request.php [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <?php
 
-require_once '../functions.php';
-require_once '../classes/Request.php';
+namespace src\classes;
```
Comment (yderevianko-cartera) [inline src/oop/pages/request.php]: namespace should be added only to classes/interface and not to templates.

File: src/oop/pages/session.php [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <?php
 
-require_once '../functions.php';
-require_once '../classes/Request.php';
+namespace src\classes;
```
Comment (yderevianko-cartera) [inline src/oop/pages/session.php]: namespace should be added only to classes/interface and not to templates.

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**